### ISSUE TITLE
EXP: Soft Label v2 설계 및 비교 실험

### ISSUE THREAD

**Issue by coouir** [LighTrip/LighTrip-AI#124]
# EXP: Soft Label v2 설계 및 비교 실험

## 📌 목적

현재 soft label v1은 baseline 학습용으로는 정상 동작하지만,  
target distribution이 다소 부드럽게 형성되며 `ivory / black / cream` 계열 색상으로 top1 추천이 집중되는 경향이 있습니다.

Soft label v2를 별도로 설계 및 생성하여 기존 v1과 비교 실험을 수행하고,  
제목 색상 추천 다양성과 추천 품질 개선 여부를 검증합니다.

---

### 📊 Soft Label v1 분석

- [ ] 현재 v1 target distribution entropy 분석
- [ ] top1 color distribution 분석
- [ ] max color share 분석
- [ ] WCAG pass tendency 분석
- [ ] 특정 색상 편향 여부 분석
- [ ] preview 결과 샘플 분석

---

### ⚙️ Soft Label v2 설계

- [ ] v2 labeling config 추가
- [ ] v2 weighting 전략 설계
- [ ] temperature 재설정
- [ ] fail penalty 조정
- [ ] readability/aesthetic balance 조정
- [ ] target distribution sharpness 비교

---

### 🏷 v2 Label 생성

- [ ] v2 pseudo-score 계산
- [ ] v2 softmax target distribution 생성
- [ ] label matrix 저장
- [ ] distribution normalization 검증
- [ ] entropy 계산 및 저장

---

### 🤖 동일 조건 학습 비교

- [ ] 동일 train/val/test split 사용
- [ ] 동일 모델 구조 사용
- [ ] 동일 optimizer/scheduler 사용
- [ ] 동일 epoch 및 batch size 유지
- [ ] v1 학습 수행
- [ ] v2 학습 수행

---

### 📈 결과 비교 분석

- [ ] train loss 비교
- [ ] val loss 비교
- [ ] val NDCG@5 비교
- [ ] test NDCG@5 비교
- [ ] top1 WCAG pass rate 비교
- [ ] top5 WCAG pass rate 비교
- [ ] top1 color distribution 비교
- [ ] max color share 비교
- [ ] entropy 분포 비교
- [ ] preview 품질 비교

---

## ⚙️ v2 후보 설정

```yaml
labeling:
  temperature: 0.15

  weights:
    readability_score: 0.65
    aesthetic_prior: 0.15
    tone_match_score: 0.15
    simplicity_score: 0.05

  fail_penalty: 0.35
```

---

## 📊 비교 지표

```text
train_loss
val_loss
val_ndcg@5
test_ndcg@5

top1_wcag_pass_rate
top5_wcag_pass_rate

max_color_share
top1_color_distribution

target_distribution_entropy

preview_quality
```

---

## 📁 산출물

```text
data/title_color_recommendation/processed/labels/labels_soft_v2.csv
data/title_color_recommendation/processed/labels/labels_matrix_v2.npy

outputs/reports/soft_label_v2_analysis.md
outputs/reports/soft_label_v1_vs_v2_comparison.md

outputs/reports/soft_label_v2_preview/
```

---

### 모델 구조

```text
4-channel ResNet18

Input:
  ROI RGB 3-channel
  + text mask 1-channel
```

### Loss

```text
KL Divergence Loss
```

### 비교 원칙

- v1 / v2는 동일한 train/val/test split 사용
- 모델 구조 및 학습 hyperparameter는 동일하게 유지
- 비교 변수는 soft label 구성만 변경
- 단순 loss만으로 판단하지 않음
- NDCG@5, WCAG pass rate, color distribution, preview 결과를 함께 분석

---

## ✅ 완료 기준

- [ ] v2 soft label이 정상 생성된다.
- [ ] 각 target distribution 합이 1에 가깝다.
- [ ] entropy 계산이 정상 동작한다.
- [ ] v1 대비 label 분포 변화가 분석된다.
- [ ] 동일 조건 학습 결과 비교가 완료된다.
- [ ] 추천 preview 비교가 완료된다.
- [ ] v2가 추천 품질 개선에 도움이 되는지 판단한다.
- [ ] 최종 soft label 사용 여부를 기록한다.

---

## ⚠️ 주의 사항

- 기존 v1 label은 삭제하지 않는다.
- v1/v2 결과를 동시에 유지한다.
- train/val/test split은 반드시 동일하게 유지한다.
- 모델 구조 및 optimizer 설정은 변경하지 않는다.
- 특정 색상으로의 collapse 여부를 반드시 확인한다.
- preview 기반 정성 평가도 함께 수행한다.
